api/rbdigital.py

```diff
@@ -704,20 +704,29 @@ def patron_remote_identifier(self, patron):
         # Set up a refresher method that takes no arguments except the
         # credential -- this is what lookup() expects.
         def refresh_credential(credential):
-            remote_identifier = self._find_or_create_remote_account(
-                patron
-            )
-            credential.credential = remote_identifier
-            credential.expires = None
+            try:
+                remote_identifier = self._find_or_create_remote_account(
+                    patron
+                )
+                credential.credential = remote_identifier
+                credential.expires = None
+            except CirculationException:
+                self.log.exception("Failed to create RBDigital User.")
             return credential
 
         # Find or create the credential.
         _db = Session.object_session(patron)
+        collection = Collection.by_id(_db, id=self.collection_id)
```

Comment:
This was where the collection session conflict was coming from. Using `self._db` by calling self.collection was giving me a different session. This resolves that issue.

Reply:
Added a better comment here for more clarity.